etna/models/nn/timesfm.py

```diff
@@ -0,0 +1,376 @@
+
+        return static_reals + static_categoricals + time_varying_reals + time_varying_categoricals
+
+    def forecast(
+        self,
+        ts: TSDataset,
+        prediction_size: int,
+        return_components: bool = False,
+    ) -> TSDataset:
+        """Make autoregressive forecasts.
+
+        Parameters
+        ----------
+        ts:
+            Dataset with features.
+        prediction_size:
+            Number of last timestamps to leave after making prediction.
+            Previous timestamps will be used as a context.
+        return_components:
+            If True additionally returns forecast components.
+
+        Returns
+        -------
+        :
+            Dataset with predictions.
+
+        Raises
+        ------
+        NotImplementedError:
+            if return_components mode is used.
+        ValueError:
+            if dataset doesn't have any context timestamps.
+        ValueError:
+            if there are NaNs in the middle or end of the time series.
+        NotImplementedError:
+            if forecasting is done without exogenous features and dataset has None frequency.
+        """
+        if return_components:
+            raise NotImplementedError("This mode isn't currently implemented!")
+
+        max_context_size = len(ts.index) - prediction_size
+        if max_context_size <= 0:
+            raise ValueError("Dataset doesn't have any context timestamps.")
+
+        if max_context_size < self.context_size:
+            warnings.warn("Actual length of a dataset is less that context size. All history will be used as context.")
+
+        self.tfm._set_horizon(prediction_size)
+
+        end_idx = len(ts.index)
+
+        all_exog = self._exog_сolumns()
+        df_slice = ts.df.loc[:, pd.IndexSlice[:, all_exog + ["target"]]]
+        first_valid_index = (
+            df_slice.isna().any(axis=1).idxmin()
+        )  # If all timestamps contains NaNs, idxmin() returns the first timestamp
+
+        target_df = df_slice.loc[first_valid_index : ts.index[-prediction_size - 1], pd.IndexSlice[:, "target"]]
+
+        nan_segment_mask = target_df.isna().any()
+        if nan_segment_mask.any():
+            nan_segments = nan_segment_mask.loc[:, nan_segment_mask].index.get_level_values(0).tolist()
+            raise ValueError(
+                f"There are NaNs in the middle or at the end of target. Segments with NaNs: {reprlib.repr(nan_segments)}."
+            )
+
+        future_ts = ts.tsdataset_idx_slice(start_idx=end_idx - prediction_size, end_idx=end_idx)
+
+        if len(all_exog) > 0:
+            target = target_df.values.swapaxes(1, 0).tolist()
+
+            exog_df = df_slice.loc[first_valid_index:, pd.IndexSlice[:, all_exog]]
+
+            nan_segment_mask = exog_df.isna().any()
+            if nan_segment_mask.any():
+                nan_segments = nan_segment_mask.loc[:, nan_segment_mask].index.get_level_values(0).tolist()
```

Comment:
I think we should do:
```
nan_segments = nan_segment_mask.loc[:, nan_segment_mask].index.get_level_values("segment").unique().tolist()
```

Reply:
Can segments be repeated?

Reply:
Here we have `MultiIndex`, the segment values will be repeated if we have exogs.